pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -166,3 +164,49 @@ func envMap(env []string) map[string]string {
 
 	return kv
 }
+
+// fromRequiredArtifacts replaces the provided builder and run images with built images from the required artifacts if specified.
+func fromRequiredArtifacts(artifact *latest.BuildpackArtifact, r ArtifactResolver, deps []*latest.ArtifactDependency, pushImages bool) (string, string, bool) {
+	builderImage, runImage := artifact.Builder, artifact.RunImage
+	builderImageLocal, runImageLocal, noPull := false, false, false
```

Comment:
nit: having this be called noPull makes readability a bit worse imo. Could be named pull and the logic flipped so that !pull is the same as noPull, and makes things easier to read in cases like !noPull

Reply:
done